src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/XGBoostEvidenceFilter.java

```diff
@@ -0,0 +1,505 @@
+        UnscaledOverlapInfo(final int numOverlap, final int numCoherent, final int overlapMappingQuality,
+                            final int coherentMappingQuality) {
+            this.numOverlap = numOverlap;
+            this.numCoherent = numCoherent;
+            this.overlapMappingQuality = overlapMappingQuality;
+            this.coherentMappingQuality = coherentMappingQuality;
+        }
+    }
+
+    /**
+     * Class that takes raw overlap info and automatically scales it to meanGenomeCoverage, storing the result for later retrieval
+     */
+    private static class CoverageScaledOverlapInfo {
+        final double numOverlap;
+        final double overlapMappingQuality;
+        final double meanOverlapMappingQuality;
+        final double numCoherent;
+        final double coherentMappingQuality;
+
+        CoverageScaledOverlapInfo(final int numOverlap, final int numCoherent, final int overlapMappingQuality,
+                                  final int coherentMappingQuality, final double meanOverlapMappingQuality,
+                                  final double coverage) {
+            this.numOverlap = ((double)numOverlap) / coverage;
+            this.overlapMappingQuality = ((double)overlapMappingQuality) / coverage;
+            this.numCoherent = ((double)numCoherent) / coverage;
+            this.coherentMappingQuality = ((double)coherentMappingQuality) / coverage;
+            this.meanOverlapMappingQuality = meanOverlapMappingQuality;
+        }
+    }
+
+    private static class CigarQualityInfo {
+        final int basesMatched;
+        final int referenceLength;
+
+        CigarQualityInfo(final BreakpointEvidence evidence) {
+            int numMatched = 0;
+            int refLength = 0;
+            final String cigarString = evidence.getCigarString();
+            if(cigarString != null) {
+                for (final CigarElement element : TextCigarCodec.decode(cigarString).getCigarElements()) {
+                    final CigarOperator op = element.getOperator();
+                    if (op.consumesReferenceBases()) {
+                        refLength += element.getLength();
+                        if (op.consumesReadBases()) {
+                            numMatched += element.getLength();
+                        }
+                    }
+                }
+            }
+            basesMatched = numMatched;
+            referenceLength = refLength;
+        }
+    }
+
+    private static double getGenomeIntervalsOverlap(final BreakpointEvidence evidence,
+                                                    final FeatureDataSource<BEDFeature> genomeIntervals,
+                                                    final ReadMetadata readMetadata) {
+        final SimpleInterval simpleInterval = evidence.getLocation().toSimpleInterval(readMetadata);
+        double overlap = 0.0;
+        for(final Iterator<BEDFeature> overlapperItr = genomeIntervals.query(simpleInterval);
```

Comment:
Re-reading this method, I realized that the mappability file contains intervals that overlap -- I think it is all 100mers in the reference that are not unique, right? Doesn't this mean that the value returned by this method can go over 1? Wouldn't you really want the fraction of the evidence that is overlapped by at least one feature from the genome intervals?

Reply:
That is a good catch, it didn't even occur to me that the mappability files would contain overlaps. You're correct about the intended goal for this computation. Of course xgboost is quite capable of dealing with the numbers that go over 1 (and maybe overlapping k-mers is better?) but probably fixing will yield better classification power.
    
Of course the same logic error is present in the training python code, so if you want a fix for this PR I'll need to fix the training code and train a new classifier.